include/CLI/Validators.hpp

```diff
@@ -14,17 +15,19 @@
 #include <string>
 
 // [CLI11:verbatim]
-#if(defined(__cplusplus) && __cplusplus >= 201703L) || (defined(_HAS_CXX17) && _HAS_CXX17 == 1)
-#define CLI11_CPP17
-#endif
 
 // C standard library
 // Only needed for existence checking
 #if defined CLI11_CPP17 && defined __has_include && !defined CLI11_HAS_FILESYSTEM
 #if __has_include(<filesystem>)
+// Filesystem cannot be used if targeting macOS < 10.15
```

Comment:
I think I would try the feature macro `__cpp_lib_filesystem` and make sure it is > 201703
https://en.cppreference.com/w/User:D41D8CD98F/feature_testing_macros.  I thought there was a few cases where clang defined the header but not all features in filesystem as well.

Reply:
That's not the problem here (though I can check that too, that's the first thing I tried). The problem is that on macOS, you can target different macOS versions. XCode 11 knows how to target 10.15, but the default is 10.14, since it's compiling on 10.14. The implementation of `<filesystem>` is fine and complete, but if you target < 10.14, you can't use filesystem since it has compiled components that are not available when linking.

Incomplete `<optional>` was a common issue, but I think they've been more careful about adding `<filesystem>` once it's complete.

Reply:
I have run into similar problems with <string_view> in the past.